### PR TITLE
Additional Java detection, like JD-GUI

### PR DESCRIPTION
**Pull request by sarnau**
https://github.com/ZeeZide/5GUIs/issues/11

### CODE REVIEW

**Comment by helje5** [comment]
Why the extra `JavaX`, does it serve any purpose beyond just `java` as a feature flag? (it is just a different jar format?)

**Comment by sarnau** [comment]
Different structure. There is no Java directory inside /Contents. It is in /Contents/Resources/Java, but we can as well check for the info.plist key.

**Comment by helje5** [comment]
Oh, I see, makes sense.

File: Sources/5GUIs/Model/InfoDict.swift [hunk 1]
```diff
@@ -61,5 +62,6 @@ struct InfoDict: Equatable {
     
     supportedPlatforms = dictionary["CFBundleSupportedPlatforms"] as? [ String ]
                       ?? []
+    JavaX = dictionary["JavaX"] != nil
```
Comment (helje5) [inline Sources/5GUIs/Model/InfoDict.swift:65]: What is an example value for `JavaX`?

Comment (sarnau) [reply Sources/5GUIs/Model/InfoDict.swift:65]: It is a dictionary:
```
		<key>JavaX</key>
		<dict>
			<key>MainClass</key>					<string>org.jd.gui.OsxApp</string>
			<key>JVMVersion</key>					<string>1.8+</string>
			<key>ClassPath</key>					<string>$JAVAROOT/jd-gui-1.6.6-min.jar</string>
			<key>WorkingDirectory</key>				<string>$JAVAROOT</string>
			<key>Properties</key>
			<dict>
				<key>apple.laf.useScreenMenuBar</key>
				<string>true</string>
			</dict>
			<key>VMOptions</key>					<string>-Xms512m</string>
		</dict>
```

Comment (sarnau) [reply Sources/5GUIs/Model/InfoDict.swift:65]: The app that goes with it is here: [https://github.com/java-decompiler/jd-gui](url)

Comment (sarnau) [reply Sources/5GUIs/Model/InfoDict.swift:65]: It's an Apple Java SDK application from what I found out.

**Pull request merged**